go/otgrpc/shared.go

```diff
@@ -18,6 +20,7 @@ type metadataReaderWriter struct {
 }
 
 func (w metadataReaderWriter) Set(key, val string) {
+	key = strings.ToLower(key)
```

Comment:
Besides opentracing-go's current HTTPHeaders implementation is case sensitive. This line is needed. The following is the (summrised) zipkin-go-opentracing implementation.

```
func (t *tracerImpl) Inject(sc opentracing.SpanContext, format interface{}, carrier interface{}) error {
        switch format {
        case opentracing.TextMap, opentracing.HTTPHeaders:
                carrier.Set(zipkinTraceID, sc.TraceID.ToHex())
        }
}
```

It calls `Set` method regardless of format (TextMap or HTTPHeaders). So it seems grpc-opentracing is responsible to make keys case insensitive.